go/enclave/storage/enclavedb/db_transaction.go

```diff
@@ -7,34 +7,46 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/status-im/keycard-go/hexutils"
 )
 
 // ---- Implement the geth Batch interface, re-using ideas and types from geth's memorydb.go ----
 
 // keyvalue is a key-value tuple that can be flagged with a deletion field to allow creating database write batches.
 type keyvalue struct {
-	key    []byte
-	value  []byte
-	delete bool
+	key      []byte
+	value    []byte
+	delRange *keyRange
+	delete   bool
 }
 
+type keyRange struct{ start, end []byte }
+
 type dbTxBatch struct {
 	timeout time.Duration
 	db      EnclaveDB
 	writes  []keyvalue
 	size    int
 }
 
-// put inserts the given value into the batch for later committing.
+func (b *dbTxBatch) DeleteRange(start, end []byte) error {
```

Comment:
where is this used? and when should it be used? a comment would be good, its not really clear why delete would be used instead of dropping the whole thing

Reply:
it's a new method that appeared in the underlying interface. I assume used by verkle trees for optimisation somehow